selectrum.el

```diff
@@ -825,39 +862,69 @@ currently displayed candidates."
             (window-resize
              win (- dheight wheight) nil nil 'pixelwise)))))))
 
-(defun selectrum--first-lines (candidates)
+(defun selectrum--ensure-single-lines (candidates)
   "Return list of single line CANDIDATES.
-Multiline canidates are merged into a single line."
-  (let ((onelines ()))
-    (dolist (cand candidates (nreverse onelines))
+Multi-line candidates are merged into a single line. The resulting
+single-line candidates are then shortened by replacing repeated
+whitespace and maybe truncating the result.
+
+These changes are indicated by strings defined in
+`selectrum-multiline-display-settings'."
```

Comment:
I think it is important to have the description in both places, because it is relevant to both places.

- In order to customize the use option correctly, it is important to know how the variable's values are being used.
- The function documentation string should describe what it does, because the function's actions are outside of the control of the user option. The user option only affects the appearance of the result of the function's actions. The replacing of the regexps will happen regardless of the option's value.

Does this make sense? I'm having trouble finding the right words to convey what I mean.

Reply:
Okay, makes sense to me, too. Maybe we can still improve the phrasing of the last sentence? I think it sounds better to formulate it more general: The formatting is determined by \`selectrum-multiline-display-settings'`.